src/Func.h

```diff
@@ -1529,6 +1632,9 @@ class Func {
      */
     EXPORT std::vector<Argument> infer_arguments() const;
 
+    bool operator < (const Func &other) const {
```

Comment:
Is this necessary? It's dangerous. Someone might accidentally say:

f() = g < h
instead of
f() = g() < h()

If you're putting Funcs in a map or somewhere, it's better to use a separate comparison type (e.g. see ExprCompare).